scripts/src2rst.py

```diff
@@ -91,22 +91,50 @@ def _gen_keybinding_rows(bindings):
     return header + rows
 
 
-def generate_commandline_options():
-    """Generate file including the command line options."""
+def generate_commandline_options(for_man=True):
```

Comment:
I am not quite sure I like the structure here. Some general thoughts:
- We could call `generate_commandline_options` only once and the function deals with doing both, splitting off where needed.
- The implementation for the website seems like a bit of a work-around as it parses the string which was formatted for the man page. Maybe the formatting should be done individually, and only the actual information is grabbed in a shared way?
- Currently it seems like the identical synopsis file is written twice. We could avoid this by writing to `docs/synopsis.rstsrc` and including it using `.. include:: /synopsis.rstsrc`. But,
- As you point out the formatting of bold regular text is not ideal for the website, as it shows up in a weird way. I guess we could try out using code ticks instead of bold for there?

This will probably result in quite a rehaul of this part as it was designed only for the man page. What do you think?

Reply:
The conversion from the manpage formatting to the list of tuples is indeed a bit hacked and not very nice. Are you aware of a method to get this data from the argument parser in some different format? Or what do you mean by your second point?

Reply:
This is a code snippet I came up with in a debug script. Instead of reading through the parser and formatting the arguments, we just grab everything we need and store it into a list of named tuples. We then have the "raw" information which could be formatted accordingly for the man page and website. There may be some parts of shared formatting that could also go into the `ParserArgument` class, e.g. the formatting of what is called title here.

<details><summary>Quick script to get information from the argparser object</summary><p>

```python
import typing

from vimiv import parser


class ParserArgument(typing.NamedTuple):
    """Storage class for a single command line argument."""
    group: str
    longname: str
    shortname: str
    metavar: typing.Any
    description: str

    @property
    def is_positional(self):
        return self.longname is self.shortname is None

    def title(self, formatter: str = "**"):
        ...


def get_arguments(argparser) -> typing.List[ParserArgument]:
    """Retrieve all arguments from the passed argparser."""
    def argument_from_action(group, action):
        if len(action.option_strings) == 2:
            longname, shortname = action.option_strings
        elif len(action.option_strings) == 1:
            longname = action.option_strings[0]
            shortname = None
        else:
            longname = shortname = None
        return ParserArgument(
            group=group.title,
            longname=longname,
            shortname=shortname,
            metavar=action.metavar,
            description=action.help,
        )

    return [
        argument_from_action(group, action)
        for group in argparser._action_groups
        for action in group._group_actions
    ]


def main():
    argparser = parser.get_argparser()
    arguments = get_arguments(argparser)
    # We should now have all the "raw" information which can then be formatted
    # accordingly for the different options
    # Possibly shared code could go into the ParserArgument class with arguments, e.g.
    # formatting the title
    for elem in arguments:
        print(elem)


if __name__ == "__main__":
    main()
```

</p></details>

Reply:
I have restructured the generator code based on you code snippet in 7226827ec31178e7978f66a21c81eb2e46a1a115. I hope it kind of reflects your idea. It definitely gets rid of the hacked man page to documentation fomat conversion. Let me know if you spot anything to improve :upside_down_face: 

I still do not really like the formatting of the synopsis in the documentation. I would prefers e.g. grey code box, but `::code-block:` does unfortunately not add line breaks.

Reply:
I will need a bit more time to read through the actual code, but it does look a lot better on a first glance. We may want to split this off into its own file though and import it in `src2rst` as it is getting rather involved.

Concerning the synopsis, we could let python do the wrapping for us using the stdlib `textwrap` module, e.g.
```python
    if man_format:
        return synopsis
    return "\n   ".join(textwrap.wrap(synopsis, width=79))
```

Reply:
Thanks for the hint with `textwrap`. I have added that. It looks much better now.

I did not yet split the code. I suggest that you review the code once you have time for that and tell me if you spot anything to improve/fix and if you prefer to move the cli option generation part into a separate file.